### DISCUSSION TITLE
Caption with soft outline possible?

### DISCUSSION THREAD

**Discussion by kodkuce** [ImageMagick/ImageMagick#2496] [Help] [answered]
Hi i would like to add some text to img, but i need it word wraped so i am using
caption.  So i wanted to add like soft shadow outline something like 
https://www.imagemagick.org/Usage/annotating/ **Fancy Label** , is there some option for that?

**Comment by kodkuce** [comment, marked as answer]
Solved it :)

convert \
-size 400x600 -background None -gravity center \
caption:"I LIKE IMAGEMAGICK"  -blur 0x6 \
-size 400x600 -background None -gravity center \
caption:"I LIKE IMAGEMAGICK" -composite \
out.png

**Comment by fmw42** [comment]
If your text is wider than the dimension you specify in that example, then it will wrap automatically with caption:. There is no option for auto-wrapping in -annotate

So you need to use caption: twice. First in black with -shadow and second in white.

![dragon](https://user-images.githubusercontent.com/13458218/91903880-d26f2d00-ec58-11ea-862f-71076e826ac4.gif)

```
width=`identify -format %w dragon.gif` \
convert \
\( -background 'none' -stroke black -strokewidth 2 -gravity center -size ${width}x30 \
    caption:"Faerie Dragons love hot apple pies\!" \
   -background black -stroke none -shadow 100x3+0+0 +repage \) \
\( -background 'none' -fill white -gravity center -size ${width}x30 \
    caption:"Faerie Dragons love hot apple pies\!" \) \
-compose over -composite \
dragon.gif +swap -gravity south -compose over -composite anno_caption.jpg

```

![anno_caption](https://user-images.githubusercontent.com/13458218/91903887-d56a1d80-ec58-11ea-8633-ddd25109e194.jpg)
